Prog/clean-DC.Rmd

```diff
@@ -1128,13 +1140,102 @@ singles <- singles %>%
                                   category_detail))
 ```
 
+## Merge on number of units to single addressess
+
+Supplement number of units for rental buildings.
+
+See how many are missing or 0
+```{r missing-units}
+
+singles %>% 
+  filter(category_detail == "Rental apartment building",
+         is.na(numberofunits) | numberofunits == 0) %>% 
+  nrow()
+
+```
+
+```{r read-units}
+
+units <- read_csv("L:/Libraries/RegHsg/Data/dc_apt_with_units.csv")
+
+singles1 <- singles %>% 
+  left_join(units, by = c("parcelbase_SSL" = "SSL")) %>% 
+  rename(numberofunits_mar = numberofunits.y,
+         numberofunits = numberofunits.x)
+
+# how many did not join
+nrow(units) -
+     (nrow(singles1) - nrow(anti_join(singles, units, by = c("parcelbase_SSL" = "SSL"))))
+
+# the bulk of the properties that did not join have no rental units
+singles %>% 
+  anti_join(units, by = c("parcelbase_SSL" = "SSL")) %>% 
+  count(category_detail)
+
+```
+
+Compare the two variables.
+```{r units-compare}
+
+# how many are different?
+singles1 %>% 
+  filter(!is.na(numberofunits),
+         numberofunits != 0,
+         !is.na(numberofunits_mar)) %>% 
+  mutate(units_flag = ifelse(numberofunits == numberofunits_mar,
+                             0,
+                             1)) %>% 
+  count(units_flag)
+
+# how different are they?
+singles1 %>% 
+  filter(!is.na(numberofunits),
+         numberofunits != 0,
+         !is.na(numberofunits_mar)) %>% 
+  mutate(difference = abs(numberofunits_mar - numberofunits)) %>%
+  filter(difference != 0) %>% 
+  summarize(avg_difference = mean(difference),
+            median_difference = median(difference),
+            difference_2 = sum(difference <= 2),
+            difference_100 = sum(difference >= 100),
+            max_difference = max(difference))
+
+# plot the difference
+singles1 %>% 
+  filter(!is.na(numberofunits),
+         numberofunits != 0,
+         !is.na(numberofunits_mar)) %>% 
+  mutate(difference = numberofunits_mar - numberofunits) %>%
+  filter(difference != 0) %>% 
+  ggplot() +
+  geom_histogram(mapping = aes(difference),
+                 bins = 70)
+
+```
+Replace `numberofunits` with `numberofunits_mar` when missing or 0.
```

Comment:
so this is favoring BK except when missing or zero? I guess I"d lean the other direction